tests/cfg_test.c

```diff
@@ -35,12 +35,47 @@
 
 #define TEST_CFG_HCL                  "tests/cfg.hcl"
 
+struct hcl_option {
+	char key[128];
+	int type;
+
+	int level;
+
+	struct hcl_option *childs;
+	struct hcl_option *next;
+	struct hcl_option *parent;
+
+	void *data;
+}
+
+static struct hcl_option *g_opts;
+
 static void usage(const char *prog_name)
```

Comment:
can you please remove `prog_name`, and use either `extern const char *__progname;` or `getprogname()` (the latter you used previously)